Bug when invoking build from cross

### Is this a new Bug?

- [X] I checked that the bug hasn't been reported before

### Package Name

any

### Package Version

any

### Device Model

any

### Device Architecture

x86_64

### Firmware Version

any

### What happened?

I noticed this during my work on qoriq rust toolchain but just confirmed, a recent change generates the following error when invoking build from cross:
```
spksrc@spksrc:~/qoriq-debug-update/spksrc/cross/openssl3$ make all-supported
===>  Pre-build native dependencies for parallel build
make[1]: Entering directory '/home/spksrc/qoriq-debug-update/spksrc/cross/openssl3'
===>  Downloading files for openssl3
===>    File openssl-3.1.4.tar.gz already downloaded
===>  Verifying files for openssl3
===>    Checking sha1sum of file openssl-3.1.4.tar.gz
===>    Checking sha256sum of file openssl-3.1.4.tar.gz
===>    Checking md5sum of file openssl-3.1.4.tar.gz
/home/spksrc/qoriq-debug-update/spksrc/cross/openssl3/../../distrib/openssl-3.1.4.tar.gz
===>  Processing dependencies of openssl3
make[2]: Entering directory '/home/spksrc/qoriq-debug-update/spksrc/cross/zlib'
===>  Downloading files for zlib
===>    File zlib-1.2.13.tar.gz already downloaded
===>  Verifying files for zlib
===>    Checking sha1sum of file zlib-1.2.13.tar.gz
===>    Checking sha256sum of file zlib-1.2.13.tar.gz
===>    Checking md5sum of file zlib-1.2.13.tar.gz
/home/spksrc/qoriq-debug-update/spksrc/cross/zlib/../../distrib/zlib-1.2.13.tar.gz
===>  Processing dependencies of zlib
===>  Extracting for zlib
tar -xzpf /home/spksrc/qoriq-debug-update/spksrc/cross/zlib/../../distrib/zlib-1.2.13.tar.gz -C /home/spksrc/qoriq-debug-update/spksrc/cross/openssl3/work 
===>  Patching for zlib
===>  Generating /home/spksrc/qoriq-debug-update/spksrc/cross/openssl3/work/zlib-1.2.13/-toolchain.cmake
env make --no-print-directory cmake_pkg_toolchain > /home/spksrc/qoriq-debug-update/spksrc/cross/openssl3/work/zlib-1.2.13/-toolchain.cmake 2>/dev/null;
make[2]: *** [../../mk/spksrc.cross-cmake.mk:41: /home/spksrc/qoriq-debug-update/spksrc/cross/openssl3/work/zlib-1.2.13/-toolchain.cmake] Error 2
make[2]: Leaving directory '/home/spksrc/qoriq-debug-update/spksrc/cross/zlib'
make[1]: *** [../../mk/spksrc.depend.mk:54: depend_target] Error 2
make[1]: Leaving directory '/home/spksrc/qoriq-debug-update/spksrc/cross/openssl3'
make: *** [../../mk/spksrc.cross-cc.mk:84: all-supported] Error 2
```

### Reproduction steps

see above

### Install Log

```log
see above
```


### Service Log

_No response_

### Other Logs

_No response_

--- a/mk/spksrc.cross-cc.mk
+++ b/mk/spksrc.cross-cc.mk
@@ -89,8 +89,17 @@ all-supported:
 	    env $(ENV) $(MAKE) -C ../../$$depend 2>&1 | tee --append build-$${depend%/*}-$${depend#*/}.log ; \
 	    [ $${PIPESTATUS[0]} -eq 0 ] || false ; \
 	  fi ; \
-	done ; \
+	done
+ifeq ($(strip $(SUPPORTED_ARCHS)),)
+	$(MAKE) supported-arch-error
+else
 	$(MAKE) $(addprefix supported-arch-,$(SUPPORTED_ARCHS))
+endif
+
+supported-arch-error:
+	@$(MSG) ########################################################
+	@$(MSG) ERROR - Please run make setup from spksrc root directory
+	@$(MSG) ########################################################
 
 supported-arch-%:
 	@$(MSG) BUILDING package for arch $* with SynoCommunity toolchain

